File: Point.py
class Point:
    def __init__(self, x=0, y=0):
        self.x = x
        self.y = y

    def get_x(self):
        return self.x

    def set_x(self, value):
        self.x = value

    def get_y(self):
        return self.y

    def set_y(self, value):
        self.y = value

    def distance(self, point):
        return ((self.x - point.x)**2 + (self.y - point.y)**2)**0.5

    def norm(self):
        return (self.x**2 + self.y**2)**0.5


class TestPoint:
    def test(self):
        point1 = Point()
        point2 = Point(3, 4)

        print(f"les cordonné par defaut: ({point1.get_x()}, {point1.get_y()})")
        print(f"Les cordonnée du point 2: ({point2.get_x()}, {point2.get_y()})")
        print(f"La distance entre point et Point2: {point1.distance(point2)}")
        print(f"la norme du Point2: {point2.norm()}")

test_point = TestPoint()
test_point.test()

File: Point.java
public class Point {
    private double x;
    private double y;

    public Point() {
        this(0, 0);
    }

    public Point(double x, double y) {
        this.x = x;
        this.y = y;
    }

    public double getX() {
        return x;
    }

    public void setX(double value) {
        this.x = value;
    }

    public double getY() {
        return y;
    }

    public void setY(double value) {
        this.y = value;
    }

    public double distance(Point point) {
        return Math.sqrt(Math.pow(this.x - point.x, 2) + Math.pow(this.y - point.y, 2));
    }

    public double norm() {
        return Math.sqrt(Math.pow(x, 2) + Math.pow(y, 2));
    }

    public static void main(String[] args) {
        Point point1 = new Point();
        Point point2 = new Point(3, 4);

        System.out.println("Default coordinates: (" + point1.getX() + ", " + point1.getY() + ")");
        System.out.println("Coordinates of Point2: (" + point2.getX() + ", " + point2.getY() + ")");
        System.out.println("Distance between Point1 and Point2: " + point1.distance(point2));
        System.out.println("Norm of Point2: " + point2.norm());
    }
}
